src/linear_algebra/determinant.lean

```diff
@@ -95,6 +95,112 @@ end
 
 end conjugate
 
+namespace linear_map
+
+/-! ### Determinant of a linear map -/
+
+variables {A : Type*} [integral_domain A] [module A M]
+variables {κ : Type*} [fintype κ]
+
+/-- The determinant of `linear_map.to_matrix` does not depend on the choice of basis. -/
+lemma det_to_matrix_eq_det_to_matrix [decidable_eq κ]
+  (b : basis ι A M) (c : basis κ A M) (f : M →ₗ[A] M) :
+  det (linear_map.to_matrix b b f) = det (linear_map.to_matrix c c f) :=
+by rw [← linear_map_to_matrix_mul_basis_to_matrix c b c,
+       ← basis_to_matrix_mul_linear_map_to_matrix b c b,
+       matrix.det_conj]; rw [basis.to_matrix_mul_to_matrix, basis.to_matrix_self]
+
+/-- The determinant of an endomorphism given a basis.
+
+See `linear_map.det` for a version that populates the basis non-computably.
+-/
+def det_aux : trunc (basis ι A M) → (M →ₗ[A] M) →* A :=
```

Comment:
Does it makes sense to call this `det_of_basis`?

Reply:
I see what you're getting at, though I'm not convinced: if you have a basis, you get a lot more affordances when you directly go to `det (linear_map.to_matrix b b f)`.

Reply:
I'm now realizing that this `trunc` stuff isn't nearly as useful as I'd hoped - as stated, this only helps for basis' indexed by the same type.Even if you change it to something monstrous like `trunc (Σ (ι : Type w') [fintype ι] [decidable_eq ι], basis ι A M)`, it still doesn't quantify over multiple universes, so you still can't prove `det_aux_def'` via properties of `trunc`.

Maybe it's still worth having anyway either because a) you've already written it! b) it's common to have different bases from `fin 3 -> R`.

Reply:
I have expanded the docstring with some notes on the pros and cons of the `trunc` parameter, sounds good?